src/compiler/scala/tools/nsc/PhaseAssembly.scala

```diff
@@ -29,9 +29,10 @@ trait PhaseAssembly {
    */
   def computePhaseAssembly(): List[SubComponent] = {
     require(phasesSet.exists(phase => phase.initial || phase.phaseName == DependencyGraph.Parser), "Missing initial phase")
-    if (!phasesSet.exists(phase => phase.terminal || phase.phaseName == DependencyGraph.Terminal))
-      if (phasesSet.add(terminal))
```

Comment:
`add` is always true?

Reply:
well, it's conservative and survives `-Wnonunit-statement`.